desktop/core/src/desktop/lib/thrift_util.py

```diff
@@ -424,6 +433,11 @@ def wrapper(*args, **kwargs):
         except TTransportException as e:
           err_msg = str(e)
           logging.info("Thrift saw a transport exception: " + err_msg, exc_info=False)
+          cache = caches[CACHES_HIVE_DISCOVERY_KEY]
+          if cache.get("hiveserver2"):
+            user_conn = cache.get("hiveserver2")
+            user_conn.pop(self.user.username, None)
+            cache.set("hiveserver2", user_conn)
```

Comment:
Here it seems that we are adding Hive specific changes in the generic Thrift lib. Any chance to move it to the beeswax lib instead?

Reply:
One option would be to apply a class decorator https://stackoverflow.com/questions/2237624/applying-a-decorator-to-every-method-in-a-class to `HiveServer2Dbms` at 
https://github.com/cloudera/hue/pull/1005/files#diff-05088e6a157c3c3047871f25b14f0ff1R93. There's currently 2 caches, 1 for the client https://github.com/cloudera/hue/pull/1005/files#diff-05088e6a157c3c3047871f25b14f0ff1R93 & 1 for Zookeeper https://github.com/cloudera/hue/pull/1005/files#diff-05088e6a157c3c3047871f25b14f0ff1R156. I believe both of them need to be cleared.

Reply:
For the client: 
When hs2 is disconnected, I will clear the cache of zk, and re-obtain a new cache in dbms.py. At the same time, DBMS_CACHE.pop (user.id) will clear the client's cache.